### PR TITLE
Adding Load Balancer strategy resource to Ping Access Adapter

### PR DESCRIPTION
**Pull request by mdsaudali365**

### CODE REVIEW

File: docs/resources/pingaccess_load_balancing_strategy.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Create a Load Balancing Strategy.
+
+## Example Usage
+```terraform
+resource "pingaccess_load_balancing_strategy" "demo_strategy" {
```
Comment (iwarapter) [inline docs/resources/pingaccess_load_balancing_strategy.md]: Please can you put the example resource in `pingaccess/test_cases/` as all these files are formatted with `terraform fmt` and will be part of a demonstration test kit.

File: docs/resources/pingaccess_load_balancing_strategy.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        	"cookieName": "PA_S",
+        }
+	}
+```
+
+## Argument Attributes
+
+The following arguments are supported:
+
+- [`className`](#className) - (Required) The class name of the load balancing strategy.
```
Comment (iwarapter) [inline docs/resources/pingaccess_load_balancing_strategy.md]: This should match the field in the schema `class_name`

File: pingaccess/resource_pingaccess_load_balancing_strategy.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		Schema: resourcePingAccessLoadBalancingStrategySchema(),
+	}
+}
+
+func resourcePingAccessLoadBalancingStrategySchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		className: &schema.Schema{
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validateClassNameValue,
```
Comment (iwarapter) [inline pingaccess/resource_pingaccess_load_balancing_strategy.go]: This won't work, the Load Balancing Strategies is one of the parts of the SDK and as such people can create their own classes to implement whatever logic they require, as we can never know what these names would be it's best to remove the validator.

File: pingaccess/resource_pingaccess_load_balancing_strategy.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		name: &schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+		},
+	}
+}
+
+func resourcePingAccessLoadBalancingStrategyCreate(d *schema.ResourceData, m interface{}) error {
+	svc := m.(*pingaccess.Client).HighAvailability
+	name := d.Get(name).(string)
```
Comment (iwarapter) [inline pingaccess/resource_pingaccess_load_balancing_strategy.go]: This should just be used directly in the LoadBalancingStrategyView definition i.e.
```go
Body: pingaccess.LoadBalancingStrategyView{
	Name: String(d.Get(name).(string)),
```

File: pingaccess/resource_pingaccess_load_balancing_strategy.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	className := d.Get(className).(string)
```
Comment (iwarapter) [inline pingaccess/resource_pingaccess_load_balancing_strategy.go]: Same as above.

File: pingaccess/resource_pingaccess_load_balancing_strategy.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+	config := d.Get(configuration).(string)
+	var dat map[string]interface{}
+	_ = json.Unmarshal([]byte(config), &dat)
+
+	input := pingaccess.AddLoadBalancingStrategyCommandInput{
+		Body: pingaccess.LoadBalancingStrategyView{
```
Comment (iwarapter) [inline pingaccess/resource_pingaccess_load_balancing_strategy.go]: It's worth implementing a `resourcePingAccessLoadBalancingStrategyData` function similar to:
https://github.com/iwarapter/terraform-provider-pingaccess/blob/master/pingaccess/resource_pingaccess_oauth_server.go#L150

Which returns a LoadBalancingStrategyView so you can remove the duplicated code in the `Create` and `Update` functions.

File: pingaccess/resource_pingaccess_load_balancing_strategy.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	_, err := svc.DeleteLoadBalancingStrategyCommand(input)
+	if err != nil {
+		return fmt.Errorf("Error deleting Load Balancing Strategy: %s", err)
+	}
+	return nil
+}
+
+func resourcePingAccessLoadBalancingStrategyReadResult(d *schema.ResourceData, input *pingaccess.LoadBalancingStrategyView) error {
+	setResourceDataString(d, "class_name", input.ClassName)
+	//d.Set("configuration", input.Configuration)
```
Comment (iwarapter) [inline pingaccess/resource_pingaccess_load_balancing_strategy.go]: Remove commented code.

File: docs/resources/pingaccess_load_balancing_strategy.md (hunk 1 above)
Comment (mdsaudali365) [reply docs/resources/pingaccess_load_balancing_strategy.md]: Added a new file under `pingaccess/test_cases -> load_balancing_strategy.tf`

File: pingaccess/resource_pingaccess_load_balancing_strategy.go (hunk 4 above)
Comment (mdsaudali365) [reply pingaccess/resource_pingaccess_load_balancing_strategy.go]: Created `resourcePingAccessLoadBalancingStrategyReadData`

File: pingaccess/resource_pingaccess_load_balancing_strategy.go (hunk 5 above)
Comment (mdsaudali365) [reply pingaccess/resource_pingaccess_load_balancing_strategy.go]: resourcePingAccessLoadBalancingStrategyReadData

File: pingaccess/resource_pingaccess_load_balancing_strategy.go (hunk 6 above)
Comment (mdsaudali365) [reply pingaccess/resource_pingaccess_load_balancing_strategy.go]: Created resourcePingAccessLoadBalancingStrategyReadData

File: pingaccess/resource_pingaccess_load_balancing_strategy.go (hunk 7 above)
Comment (mdsaudali365) [reply pingaccess/resource_pingaccess_load_balancing_strategy.go]: Done

**Pull request closed without merging**